otx/algorithms/action/tasks/inference.py

```diff
@@ -174,7 +184,7 @@ def dummy_dump_saliency_hook(model, inp, out):
 
         metric = None
         metric_name = self._recipe_cfg.evaluation.final_metric
-        if inference_parameters.is_evaluation:
+        if inference_parameters and inference_parameters.is_evaluation:
```

Comment:
I assume you want to check whether the variable named `inference_parameters` existed.
Then, should we check the existence of the variable first?

```
if inference_parameters: 
    if inference_parameters.is_evaluation:  
        metric = ...  
```